internal/docker-client/worker-event-loop.go

```diff
@@ -0,0 +1,16 @@
+package docker_client
+
+import (
+	"context"
+	"github.com/Capstone-auto-grader/grader-api-v2/internal/grader-task"
+	sync_map "github.com/Capstone-auto-grader/grader-api-v2/internal/sync-map"
+)
+
+func WorkerEventLoop(queue <-chan string, client *DockerClient, tasks *sync_map.SyncMap) {
+	for {
+		taskId := <-queue
```

Comment:
instead you can write `for taskId := range queue { ... }` this will end the loop when the channel is closed by the sender, avoiding an infinite loop.